src/application/services/useEditor.ts

```diff
@@ -0,0 +1,181 @@
+import { onBeforeUnmount, onMounted, ref, watch, type Ref } from 'vue';
+import Editor, { type OutputData, type API } from '@editorjs/editorjs';
+import type EditorTool from '@/domain/entities/EditorTool';
+// @ts-expect-error: we need to rewrite plugins to TS to get their types
+import Header from '@editorjs/header';
+import { useAppState } from './useAppState';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type GlobalEditorTool = any;
+
+/**
+ * Interface describing hook props
+ */
+interface EditorProps {
+  /**
+   * Content displayed in Editor
+   */
+  content?: OutputData;
+
+  /**
+   * Allow edit content in Editor
+   */
+  isReadOnly?: boolean;
+
+  /**
+   * Handler for every content change in Editor
+   */
+  onChange: (api: API) => void;
+}
+
+/**
+ * Add promise which is rejected after loading is complete
+ *
+ * @param src - source path to tool
+ */
+function loadScript(src: string): Promise<Event> {
+  return new Promise(function (resolve, reject) {
+    const script = document.createElement('script');
+
+    script.src = src;
+    script.onload = resolve;
+    script.onerror = reject;
+    document.head.appendChild(script);
+  });
+}
+
+/**
+ * Load user editor tools
+ *
+ * @param userEditorTools User editor tools
+ */
+function loadUserTools(
```

Comment:
lets make it more universal:

```suggestion
function loadTools(toolsList)
```